pre_commit/clientlib.py

```diff
@@ -112,6 +112,24 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+class CheckMutableRev:
+    key = 'rev'
+
+    def check(self, dct: Dict[str, Any]) -> None:
+        if any(
+            mutable in dct.get(self.key, '')
+            for mutable in {'HEAD', 'stable', 'master'}
+        ):
+            logger.warning(
+                f"The {self.key!r} field of repo {dct.get('repo')!r} "
+                f"shouldn't be mutable",
+            )
+
+    apply_default = cfgv.Required.apply_default
+
+    remove_default = cfgv.Required.remove_default
```

Comment:
`rev` isn't `Required`, for example `repo: local` / `repo: meta` don't use it

Reply:
didn't think of that, changed it to `Conditional`